test/NuGet.Core.Tests/NuGet.Commands.Test/SignCommandRunnerTests.cs

```diff
@@ -109,7 +109,8 @@ public async Task ExecuteCommandAsync_WithAmbiguousMatch_ThrowsAsync()
             using (var test = await Test.CreateAsync(_fixture.GetDefaultCertificate()))
             {
                 test.Args.CertificateSubjectName = "Root";
-                test.Args.CertificateStoreLocation = CertificateStoreUtilities.GetTrustedCertificateStoreLocation();                
+                //X509 store is opened in ReadOnly mode in this code path. Hence StoreLocation is set to LocalMachine.
+                test.Args.CertificateStoreLocation = StoreLocation.LocalMachine;                
```

Comment:
I kept looking at this code, and I have a hunch that this sort of thing is better off in an overload of `CertificateStoreUtilities.GetTrustedCertificateStoreLocation()` that allows you to do `CertificateStoreUtilities.GetTrustedCertificateStoreLocation(readOnly: true)` instead, and hide the StoreLocation logic behind that method, as it did before. What do you think of that approach?